fees/dependencies.go

```diff
@@ -0,0 +1,11 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fees
+
+type Metadata interface {
```

Comment:
Not sure here if we want a more useful name (or if it suffices)

Reply:
I don't think we should use `Metadata` for this...not sure what makes sense yet.

I would call this interface `Fees` and keep it in `chain/dependencies`?

Reply:
The only problem that I'm facing is that `computeNext` takes in `Rules` as a parameter, which could lead to an import cycle. I'm not really sure how to work around that